### PR TITLE
Semantic Release setup

### PR DESCRIPTION
**Pull request by mmazzarolo**
- [x] Added the `semantic-release` and `@semantic-release/git` packages
- [x] Created a standard `.releaserc` file for configuring the release process
- [x] Added a `CONTRIBUTING.md` to describe the contributing release process
- [x] Added a CircleCI config file (will be updated later with the `orb` once we have a test suite)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - setup
+      - tests:
+          requires:
+            - setup
+      - publish:
+          requires:
+            - tests
+          filters:
+            branches:
+              only: master
```
Comment (matt-oakes) [inline .circleci/config.yml]: I would recommend changing the CircleCI config to the following. The reason being that you can't guarantee that the installed `node_modules` will work correctly between jobs. You are restoring based on the architecture, however, in the `tests` and `publish` steps there's not guarantee that they will be installed if the cache is not hit (different arch, or just CircleCI having a bad day).

The update I propose uses a CircleCI Orb which includes the correct logic for saving and restoring the yarn Cache.

```yaml
version: 2.1

orbs:
  rn: react-native-community/react-native@1.1.0

defaults: &defaults
  docker:
    - image: circleci/node:10

jobs:
  checkout_code:
    <<: *defaults
    steps:
      - checkout
      - persist_to_workspace:
          root: .
          paths: .
  
  test:
    <<: *defaults
    steps:
      - attach_workspace:
          at: .
      - rn/yarn_install
      - run:
        name: Run Tests
        command: yarn test
  
  publish:
    <<: *defaults
    steps:
      - attach_workspace:
          at: .
      - rn/yarn_install
      - run:
          name: Publish to NPM
          command: yarn run release

workflows:
  version: 2
  ci:
    jobs:
      - checkout_code
      - test:
          requires:
            - setup
      - publish:
          requires:
            - test
          filters:
            branches:
              only: master
```

File: .releaserc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "plugins": [
+    "@semantic-release/commit-analyzer",
+    "@semantic-release/release-notes-generator",
+    "@semantic-release/npm",
+    "@semantic-release/github",
+    "@semantic-release/git"
```
Comment (matt-oakes) [inline .releaserc]: I suggest customising the commit message a little bit to match the other repos using Semantic Release.

```suggestion
    [
      "@semantic-release/git",
      {
        "assets": "package.json",
        "message": "chore(release): ${nextRelease.version} [skip ci] \n\n${nextRelease.notes}"
      }
    ]
```

Comment (mmazzarolo) [reply .releaserc]: I checked and it seems to be like the default https://github.com/semantic-release/git
But will change it for the sake of consistency, good catch!

File: .circleci/config.yml (hunk 1 above)
Comment (mmazzarolo) [reply .circleci/config.yml]: Awesome! 🙏

**Pull request merged**